docs/developer/decision-records/2022-03-01-repo-split/README.md

```diff
@@ -0,0 +1,230 @@
+# EDC Repository Split
+
+Currently, the EDC codebase is stored in a monorepo in Github where core modules coexist with vendor-specific extensions. This strategy allows for flexibility especially during the initial phase of development where bigger application-wide refactorings are common. Integration tests spanning several extensions are easy to write and help to reduce the likelihood of a regression within a certain module affecting other modules. 
+
+As more contributors join the project and more vendor-specific implementations are added to the codebase, the drawbacks of a monorepo become apparent:
+
+- A new version of an SPI forces an immediate adaptation of all implementing modules. This requires coordination among several teams, ultimately slowing down the development cycle.
+- A new version of a module or extension requires a full EDC release, even for small fixes.
+- A full test suite run for any change is suboptimal. Having separate smaller modules improves CI performance, but this can be achieved as well on a monorepo by having multiple parallel CI pipelines.
+- Danger of having too much vendor-specific code within EDC core, giving the false impression of not being a 100% cloud-agnostic project.
+- Danger of having too many contributors to the EDC Core functionality leading to misalignment and inconsistencies.
+
+## EDC multirepo strategy
+
+### By vendor
```

Comment:
Feedback from @algattik:
> In the ADR we should mention Conway's law, and how governance has to be aligned with these open source repos. Seems this really forces the by-vendor approach

Reply:
@algattik I added a point about Conway's law. Please take look.
For the second part of your feedback
> how governance has to be aligned with these open source repos. Seems this really forces the by-vendor approach
Could you elaborate on that? What exactly should be included in the spike doc?